docs/deployment/configuration/auth_setup.rst

```diff
@@ -597,7 +601,8 @@ Follow the steps in this section to configure `flyteadmin` to use an external au
    
          secrets:
            adminOauthClientCredentials:
-             enabled: true # see the section "Disable Helm secret management" if you require to do so
+             enabled: true
+             create: true # see the section "Disable Helm secret management" if you require to do so
```

Comment:
Thanks for your contribution and sorry for the inconvenience so far.

To make it as clear as possible to users, the challenge is that right now `create` here means both `create` AND `mount`.

What we should try to do is to decouple those two actions. So `enabled` should become `mount` and be used as a conditional for the volumeMount, with `create` used as conditional for the `secret-auth` template as you're proposing.

There would be an unsupported combination tho: `mount.false` and `create.true`.

The use case of an external Secrets Manager would be handled with `mount.create` and `create.false`

The default for auth would be both `true` and defaults without `auth` both `false`.

What do you think?

Reply:
No worries, glad that I got chance to contribute! 
I totally agree that using `create` and `mount` as key would be more intuitive, will modify it! 
BTW I’m wondering if it’s necessary to warn the user if they use the unreasonable combination, the most low-hanging way of doing it would be adding comments in the `value.yaml`, any other suggestions?

Reply:
I'd go with the comment for now to warn users